src/StoreKeeper/WooCommerce/B2C/Imports/LocationImport.php

```diff
@@ -0,0 +1,725 @@
+        }
+
+        if($wpdb->query(OpeningSpecialHoursModel::prepareQuery($deleteQuery))) {
+            $this->debug(
+                sprintf(
+                    'The old special opening hours of location id=%d were successfully removed.',
+                    $locationStoreKeeperId
+                )
+            );
+        }
+
+        return $this;
+    }
+
+    /**
+     * Get location opening hours mapped by day of week
+     *
+     * @param int $locationId
+     * @param bool $reload
+     * @return array
+     */
+    protected function getLocationOpeningHours(int $locationId, bool $reload = false)
+    {
+        if (!array_key_exists($locationId, $this->locationOpeningHours) || $reload) {
+            $openingHours = OpeningHourModel::getByLocationId($locationId);
+
+            $this->locationOpeningHours[$locationId] = array_combine(
+                array_column(
+                    $openingHours,
+                    'open_day'
+                ),
+                $openingHours
+            );
+
+            unset($openingHours);
+        }
+
+        return $this->locationOpeningHours[$locationId];
+    }
+
+    /**
+     * Get location special opening hours ID map (`storekeeper_id` -> `id`)
+     *
+     * @param int $locationId
+     * @param bool $reload
+     * @return array
+     */
+    protected function getLocationSpecialOpeningHoursIdMap(int $locationId, bool $reload = false)
+    {
+        global $wpdb;
+
+        if (!array_key_exists($locationId, $this->locationSpecialOpeningHoursIdMap) || $reload) {
+            $mapSelectQuery = OpeningSpecialHoursModel::getSelectHelper()
+                ->cols([OpeningSpecialHoursModel::PRIMARY_KEY, 'storekeeper_id'])
+                ->where('location_id = :location_id')
+                ->bindValue('location_id', $locationId);
+
+            $this->locationSpecialOpeningHoursIdMap[$locationId] = array_column(
+                $wpdb->get_results(OpeningSpecialHoursModel::prepareQuery($mapSelectQuery)),
+                OpeningSpecialHoursModel::PRIMARY_KEY,
+                'storekeeper_id'
+            );
+
+            unset($mapSelectQuery);
+        }
+
+        return $this->locationSpecialOpeningHoursIdMap[$locationId];
+    }
+
+    /**
+     * Get mapped days of week (numeric representation -> full textual representation)
+     *
+     * @return array
+     */
+    protected function getDaysOfWeek(): array
+    {
+        if (null === $this->daysOfWeek) {
+            $oldLocale = setlocale(LC_TIME, '0');
+            $dayStart = date('d', strtotime('next Monday'));
+            $this->daysOfWeek = [];
+
+            setlocale(LC_TIME, 'en_US');
+            for ($i = 0; $i < 7; $i++) {
+                $time = mktime(0, 0, 0, date('m'), $dayStart + $i, date('y'));
+
+                $this->daysOfWeek[date('w', $time)] = strtolower(date('l', $time));
+            }
+            setlocale(LC_TIME, $oldLocale);
+        }
+
+        return $this->daysOfWeek;
```

Comment:
> I think this should be enough, `open_day` is hardcoded in backend. Just kindly add as a comment why we did these changes above.
> 
> Here is the reference in backend: ![image](https://private-user-images.githubusercontent.com/18332309/393216005-2c383936-38fe-48ad-ba5f-9b99696439cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3MzUwOTIsIm5iZiI6MTczMzczNDc5MiwicGF0aCI6Ii8xODMzMjMwOS8zOTMyMTYwMDUtMmMzODM5MzYtMzhmZS00OGFkLWJhNWYtOWI5OTY5NjQzOWNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA5VDA4NTk1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMzZTI4ZGJlZTgwN2Y1OGI1MWIyMjFkMzAwNjY0ZGIyMGQ1N2M0ZGRmYTQ3YmIxZDk5Mzk5ZGU1ZWZlMTMwZGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.19V8hEUjlc6IA4NK-KW1_0-7ocYhxJ8jQCMYvvEkdCE)

we did in this way to map the numerical representation of the day of the week to the textual representation, as we save in DB the day of the week as numeric representation (_0_ for _Sunday_, _1_ for _Monday_ and so on (https://www.php.net/manual/en/datetime.format.php -> see _w_ format character)) (see: https://github.com/storekeeper-company/storekeeper-woocommerce-b2c/blob/eaef6747937ceb36afd69c06a9af244d23ca6323/src/StoreKeeper/WooCommerce/B2C/Migrations/Versions/V20241113000020LocationOpeningHour.php#L32) because we wanted to use the Wordpress style, as it will be easier, for example, at translation https://developer.wordpress.org/reference/classes/wp_locale/get_weekday/ as here https://github.com/storekeeper-company/storekeeper-woocommerce-b2c/blob/eaef6747937ceb36afd69c06a9af244d23ca6323/src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/LocationsTab.php#L351 and so on.